pallets/cash/src/lib.rs

```diff
@@ -329,6 +359,53 @@ impl<T: Config> pallet_session::SessionManager<SubstrateId> for Module<T> {
     }
 }
 
+/*
+-- Block N --
+changeAuth extrinsic, nextValidators set, pendingEraNotices set, IsEraChangePending set true
+
+-- Block N + 1 --
+Must end session to get through the already-planned session
+* "ShouldEndSession" should be true bc nextValidators not empty
+* "new_session" returns nextValidators to queue them and deletes the storage
+
+-- Afterwards --
+"ShouldEndSession" when
+* era notices were signed
+
+The next session will finally have the NextValidators, and notice signing can continue
+
+*/
+
+// periodic except when new authorities are pending and when an era notice has just been completed
+impl<T: Config> pallet_session::ShouldEndSession<T::BlockNumber> for Module<T> {
+    fn should_end_session(now: T::BlockNumber) -> bool {
+        if NextValidators::iter().count() > 0 {
+            // we have not planned the new auth session yet
+            true
+        } else if IsEraChangePending::get() == true && PendingEraNotices::iter().count() == 0 {
+            // era change has just occured and we are ready to switch sessions
+            IsEraChangePending::put(false);
+            true
+        } else {
+            // no era changes pending, periodic
+            let period: T::BlockNumber = <T>::BlockNumber::from(params::SESSION_PERIOD as u32);
+            (now % period) == <T>::BlockNumber::from(0 as u32)
+        }
+    }
+}
+
+impl<T: Config> frame_support::traits::EstimateNextSessionRotation<T::BlockNumber> for Module<T> {
+    fn estimate_next_session_rotation(now: T::BlockNumber) -> Option<T::BlockNumber> {
```

Comment:
is this just part of the interface, or used somehow?

Reply:
its just part of the interface, not used for anything afaict